batchflow/research/experiment.py

```diff
@@ -754,31 +756,46 @@ def close_logger(self):
 
     def call(self, name, iteration, n_iters=None):
         """ Execute one iteration of the experiment. """
-        if self.is_alive or name.startswith('__'):
-            if self._profiler:
-                self._profiler.enable()
-
-            self.last = self.last or (iteration + 1 == n_iters)
-            self.iteration = iteration
-
-            self.logger.debug(f"Execute '{name}' [{iteration}/{n_iters}]")
-            exception = (StopIteration, KeyboardInterrupt) if self.debug else Exception
-            try:
-                self.outputs[name], unit_time = self.actions[name](iteration, n_iters, last=self.last)
-            except exception as e: #pylint:disable=broad-except
-                self.is_failed = True
-                self.last = True
-                if isinstance(e, StopIteration):
-                    self.logger.info(f"Stop '{name}' [{iteration}/{n_iters}]")
-                else:
-                    self.exception = e
-                    ex_traceback = e.__traceback__
-                    msg = ''.join(traceback.format_exception(e.__class__, e, ex_traceback))
-                    self.logger.error(f"Fail '{name}' [{iteration}/{n_iters}]: Exception\n{msg}")
-                    if self.monitor:
-                        self.monitor.fail_item_execution(name, self, msg)
-            if self.is_failed and ((list(self.actions.keys())[-1] == name) or (not self.executor.finalize)):
-                self.is_alive = False
+        if self.redirect_stdout in [2, 3] and self.dump_results:
+            if self.redirect_stdout == 1:
```

Comment:
You will never enter this condition, as you checked that `self.redirect_stdout in [2, 3]`